src/main/java/org/mskcc/cmo/messaging/impl/NATSGatewayImpl.java

```diff
@@ -37,6 +40,9 @@
     @Value("${nats.url}")
     private String natsURL;
 
+    @Autowired
+    FileUtil fileUtil;
```

Comment:
so this would have to be managed by spring in order to be autowired

you can add `@Component` to the implementation of the file util class.

Reply:
Be sure to test these new changes also. I think if you had built and tried invoking the file writer that the 1. filepath property wouldnt have been able to inject into the writer class and 2. the application context wouldnt have been able to find a bean of type `FileUtil`